### PR TITLE
Fix GCC 5.x warnings

### PR DESCRIPTION
**Pull request by mshabunin**
Fixing several GCC 5.x warnings.

### CODE REVIEW

File: 3rdparty/libtiff/CMakeLists.txt [hunk 1]
```diff
@@ -95,7 +95,7 @@ ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4267 /wd4305 /wd4306) # vs2008 Win64
 ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4703) # vs2012
 ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4456 /wd4457 /wd4312) # vs2015
 
-ocv_warnings_disable(CMAKE_C_FLAGS /wd4267 /wd4244 /wd4018)
+ocv_warnings_disable(CMAKE_C_FLAGS /wd4267 /wd4244 /wd4018 /wd4311 /wd4312)
```
Comment (alalek) [inline 3rdparty/libtiff/CMakeLists.txt]: > /wd4311 /wd4312

I believe "uintptr_t/intptr_t" should be used in these cases.

>

File: modules/core/perf/perf_umat.cpp [hunk 2]
```diff
@@ -13,7 +13,6 @@ using namespace ::testing;
 using std::tr1::tuple;
 using std::tr1::get;
 
-namespace {
```
Comment (alalek) [inline modules/core/perf/perf_umat.cpp]: What is kind of problem here with anonymous namespace?

**Comment by alalek** [comment]
:+1:

File: modules/core/perf/perf_umat.cpp (hunk 2 above)
Comment (mshabunin) [inline modules/core/perf/perf_umat.cpp]: In this specific case after `OCL_PERF_TEST_P` macro expansion unused variable ( opencv/modules/ts/include/opencv2/ts/ts_gtest.h:17036 file, INSTANTIATE_TEST_CASE_P macro, int variable with _dummy_ suffix) caused a warning.

File: 3rdparty/libtiff/CMakeLists.txt (hunk 1 above)
Comment (mshabunin) [inline 3rdparty/libtiff/CMakeLists.txt]: This is for libtiff only, I decided not to change code in 3rdparty library. These warning were produced by VS2015 during  `opencv/3rdparty/libtiff/tif_win32.c`  file compilation.

**Pull request merged**